packages/backend/src/projectAdapters/dbtBaseProjectAdapter.ts

```diff
@@ -17,44 +23,70 @@ export abstract class DbtBaseProjectAdapter implements ProjectAdapter {
 
     public async compileAllExplores(
         loadSources: boolean = false,
-    ): Promise<Explore[]> {
-        // Install dependencies for dbt and fetch the manifest
+    ): Promise<(Explore | ExploreError)[]> {
+        // Install dependencies for dbt and fetch the manifest - may raise error meaning no explores compile
         await this.rpcClient.installDeps();
         const { manifest } = await this.rpcClient.getDbtManifest();
 
-        // Validate models in the manifest - may throw ParseError
-        const models = await DbtBaseProjectAdapter._validateDbtModels(
-            Object.values(manifest.nodes).filter(
-                (node) => node.resource_type === 'model',
-            ) as DbtModelNode[],
+        // Validate models in the manifest - models with invalid metadata will not compile
+        const models = Object.values(manifest.nodes).filter(
+            (node) => node.resource_type === 'model',
+        ) as DbtModelNode[];
+        const validatedModels = models.map(
+            DbtBaseProjectAdapter._isDbtModelMetadataValid,
         );
-
         const adapterType = manifest.metadata.adapter_type;
 
-        // Be lazy and try to type the models without refreshing the catalog
+        // Models with invalid metadata compile to failed explores
+        const failedExplores: ExploreError[] = validatedModels
+            .filter(
+                (
+                    value,
+                ): value is {
+                    isValid: true;
+                    model: DbtModelNode;
+                    errorMessage: string;
+                } => !value.isValid,
+            )
+            .map((value) => ({
+                name: value.model.name,
+                errors: [
+                    { type: 'MetadataParseError', message: value.errorMessage },
+                ],
+            }));
+
+        // Be lazy and try to attach types to the remaining models without refreshing the catalog
+        const validModels = validatedModels
+            .filter((v) => v.isValid)
+            .map((v) => v.model);
```

Comment:
We are looping 3 times over the same data. 
The validate function could handle this in a single loop. 

```
const [validModels, invalidModels] = DbtBaseProjectAdapter._validateDbtModels(models);
```

```
(models) => {

    return models.reduce(([valid, invalid], model) => {
         const model = ... 
         
         if (model.isValid) {
              return [[...valid, model], invalid];
         } else {
             return [valid, [...invalid, model]];
         }
    }, [[],[]]);
}
```